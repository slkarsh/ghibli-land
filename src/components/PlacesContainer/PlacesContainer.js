import React from 'react'
import { connect } from 'react-redux'
import PlacesCard from '../PlacesCard/PlacesCard'
import './PlacesContainer.scss'

export const PlacesContainer = ({ places }) => {
  const displayPlaces = places.map((place, index) => {
    return (
      <PlacesCard 
        {...place}
        key={index}
      />
    )
  })

  return (
    <section className='places-container'>
      {displayPlaces}
    </section>
  )
}

const mapStateToProps = state => ({
  places: state.places
})

export default connect(mapStateToProps)(PlacesContainer)