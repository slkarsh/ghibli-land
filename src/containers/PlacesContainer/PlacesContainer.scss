.places-container {
  display: grid;
  grid-gap: 20px 15px;
  padding: 20px 20px;
  grid-template-columns: repeat(auto-fit, minmax(300px, 0.5fr));
  overflow: scroll;
}