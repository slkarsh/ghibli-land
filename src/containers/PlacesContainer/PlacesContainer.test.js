import React from 'react'
import { PlacesContainer, mapStateToProps } from './PlacesContainer'
import { shallow } from 'enzyme'

describe('PlacesContainer', () => {
  describe('PlacesContainer component', () => {
    let wrapper;
    beforeEach(() => {
      const mockCheckPlaces = jest.fn()
      const mockPlaces = [
        {
          name: 'the moon',
          climate: 'cold',
          terrain: 'rocky'
        },
        {
          name: 'the earth',
          climate: 'variable',
          terrain: 'earthy'
        }
      ]
      wrapper = shallow(<PlacesContainer
        checkPlaces={mockCheckPlaces}
        places={mockPlaces}
      />)
    })
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
  describe('mapStateToProps', () => {
    it('should return an object with the places array', () => {
      const mockState = {
        movies: [
          {
            title: 'another film',
            description: 'it did not win an oscar',
            release_date: '2001',
            rt_score: 2
          },
          {
            title: 'a short film',
            description: 'shockingly didnt win any awards',
            release_date: '1776',
            rt_score: 5
          }
        ],
        places: [
          {
            name: 'the moon',
            climate: 'cold',
            terrain: 'rocky'
          },
          {
            name: 'the earth',
            climate: 'variable',
            terrain: 'earthy'
          }
        ]
      }
      const expectedState = {
        places: [
          {
            name: 'the moon',
            climate: 'cold',
            terrain: 'rocky'
          },
          {
            name: 'the earth',
            climate: 'variable',
            terrain: 'earthy'
          }
        ]
      }
      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expectedState)
    })
  })
})