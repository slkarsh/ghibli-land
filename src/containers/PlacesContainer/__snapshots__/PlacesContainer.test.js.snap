// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlacesContainer PlacesContainer component should match the snapshot 1`] = `
<section
  className="places-container"
>
  <Connect(PlacesCard)
    checkPlaces={[MockFunction]}
    climate="cold"
    key="0"
    name="the moon"
    terrain="rocky"
  />
  <Connect(PlacesCard)
    checkPlaces={[MockFunction]}
    climate="variable"
    key="1"
    name="the earth"
    terrain="earthy"
  />
</section>
`;
